### PR TITLE
changed all system.out and e.print to use Logger

### PR DESCRIPTION
**Pull request by sleim22**

### CODE REVIEW

File: src/main/java/com/github/s7connector/impl/S7BaseConnection.java [hunk 1]
```diff
@@ -59,13 +61,14 @@ public static void checkResult(final int libnodaveResult) {
...
-	 *            the byte stream
+	 * @param b the byte stream
 	 */
 	protected static void dump(final byte[] b) {
+		StringBuilder stringBuilder = new StringBuilder();
 		for (final byte element : b) {
-			System.out.print(Integer.toHexString(element & 0xFF) + ",");
+			stringBuilder.append(Integer.toHexString(element & 0xFF) + ",");
 		}
+		LOGGER.trace(stringBuilder.toString());
```
Comment (Zohhak) [inline src/main/java/com/github/s7connector/impl/S7BaseConnection.java]: would change to debug, missing check "isDebugEnabled"

File: src/main/java/com/github/s7connector/impl/S7BaseConnection.java [hunk 2]
```diff
@@ -59,12 +61,15 @@ public static void checkResult(final int libnodaveResult) {
...
 	 */
 	protected static void dump(final byte[] b) {
-		for (final byte element : b) {
-			System.out.print(Integer.toHexString(element & 0xFF) + ",");
+		if (LOGGER.isDebugEnabled()) {
+			StringBuilder stringBuilder = new StringBuilder();
+			for (final byte element : b) {
+				stringBuilder.append(Integer.toHexString(element & 0xFF) + ",");
+			}
+			LOGGER.trace(stringBuilder.toString());
```
Comment (Zohhak) [inline src/main/java/com/github/s7connector/impl/S7BaseConnection.java]: debug

File: src/main/java/com/github/s7connector/impl/S7TCPConnection.java [hunk 3]
```diff
@@ -15,145 +15,145 @@
...
+		this.plcType = plcType;
+		this.setupSocket();
+	}
+
+	@Override
+	public void close() {
+		try {
+			this.socket.close();
+		} catch (final Exception e) {
+			LOGGER.error("Interrupted: {}", e);
```
Comment (Zohhak) [inline src/main/java/com/github/s7connector/impl/S7TCPConnection.java]: remove {} exception messages are extended by default

File: src/main/java/com/github/s7connector/impl/nodave/Nodave.java [hunk 4]
```diff
@@ -135,19 +139,22 @@ public static byte[] bswap_8(final int a) {
...
-		System.out.print(text + " ");
-		for (int i = start; i < (start + len); i++) {
-			int j = mem[i];
-			if (j < 0) {
-				j += 256;
-			}
-			String s = Integer.toHexString(j);
-			if (s.length() < 2) {
-				s = "0" + s;
+		LOGGER.debug(text);
```
Comment (Zohhak) [inline src/main/java/com/github/s7connector/impl/nodave/Nodave.java]: move 1 line down

File: src/main/java/com/github/s7connector/impl/nodave/PLCinterface.java [hunk 5]
```diff
@@ -57,7 +61,7 @@ public int read(final byte[] b, int start, int len) {
 					}
 					retry++;
 				} catch (final InterruptedException e) {
-					e.printStackTrace();
+					LOGGER.error("Interrupted {}", e);
```
Comment (Zohhak) [inline src/main/java/com/github/s7connector/impl/nodave/PLCinterface.java]: remove {}

File: src/main/java/com/github/s7connector/impl/nodave/PLCinterface.java [hunk 6]
```diff
@@ -68,7 +72,7 @@ public int read(final byte[] b, int start, int len) {
 			}
 			return res;
 		} catch (final IOException e) {
-			e.printStackTrace();
+			LOGGER.error("Interrupted {}", e);
```
Comment (Zohhak) [inline src/main/java/com/github/s7connector/impl/nodave/PLCinterface.java]: remove {}

File: src/main/java/com/github/s7connector/impl/nodave/PLCinterface.java [hunk 7]
```diff
@@ -77,7 +81,7 @@ public void write(final byte[] b, final int start, final int len) {
 		try {
 			this.out.write(b, start, len);
 		} catch (final IOException e) {
-			System.err.println("Interface.write: " + e);
+			LOGGER.error("Interface.write: {}", e);
```
Comment (Zohhak) [inline src/main/java/com/github/s7connector/impl/nodave/PLCinterface.java]: remove {}

File: src/main/java/com/github/s7connector/impl/nodave/S7Connection.java [hunk 8]
```diff
@@ -291,7 +295,7 @@ public int readBytes(final DaveArea area, final int DBnum, final int start, fina
 		try {
 			this.semaphore.acquire();
 		} catch (final InterruptedException e) {
-			e.printStackTrace();
+			LOGGER.error("Interrupted {}", e);
```
Comment (Zohhak) [inline src/main/java/com/github/s7connector/impl/nodave/S7Connection.java]: remove {}

File: src/test/java/com/github/s7connector/test/ConnectionDropTest.java [hunk 9]
```diff
@@ -15,41 +15,41 @@
...
 
-		final int port = (int)(Math.random() * 10000) + 10000;
+		final int port = (int) (Math.random() * 10000) + 10000;
 		final ServerSocket serverSocket = new ServerSocket(port);
 		new Thread(() -> {
 			try {
 				Socket socket = serverSocket.accept();
 			} catch (IOException e) {
-				e.printStackTrace();
+				LOGGER.error("Interrupted {}", e);
```
Comment (Zohhak) [inline src/test/java/com/github/s7connector/test/ConnectionDropTest.java]: remove{}

File: src/test/java/com/github/s7connector/test/DateTest.java [hunk 10]
```diff
@@ -18,30 +18,29 @@
...
 	@Test
-	public void test2()
-	{
+	public void test2() {
 		Calendar c = Calendar.getInstance();
 		c.clear();
 		c.set(Calendar.YEAR, 1990);
-		System.out.println( c.getTime() );
-			
+		LOGGER.debug(c.getTime().toString());
```
Comment (Zohhak) [inline src/test/java/com/github/s7connector/test/DateTest.java]: toString() optional, debugger should call that internally

File: src/test/java/com/github/s7connector/test/connector/EchoConnector.java [hunk 11]
```diff
@@ -17,49 +17,51 @@
...
+				buffer.length);
 
 		this.buffer = buffer;
-
-		System.out.println("Size: " + buffer.length);
-
-		for (int i = 0; i < buffer.length; i++)
-			System.out.print(Integer.toHexString(buffer[i] & 0xFF) + ",");
-
+		LOGGER.debug("Size: {}", buffer.length);
```
Comment (Zohhak) [inline src/test/java/com/github/s7connector/test/connector/EchoConnector.java]: move 1 line down

File: src/test/java/com/github/s7connector/test/converter/IntegerConverterTest.java [hunk 12]
```diff
@@ -17,61 +17,61 @@
...
-
+		byte[] buffer = new byte[] { 0, 0, 0x02, (byte) 0x9a, 0, 0 };
+		debug(buffer);
 		int i = c.extract(Integer.class, buffer, 2, 0);
 		Assert.assertEquals(666, i);
 	}
-	
+
+	private void debug(byte[] buffer) {
+		StringBuilder stringBuilder = new StringBuilder();
```
Comment (Zohhak) [inline src/test/java/com/github/s7connector/test/converter/IntegerConverterTest.java]: isDebugEnabled

File: src/test/java/com/github/s7connector/test/live/RealPLCTest.java [hunk 13]
```diff
@@ -17,145 +17,104 @@
...
-	{
-		
+	public static void main(String[] args) throws Exception {
+
 		/*
-		 * Byte:
-		 * Tested: 2012-01-20 OK
-		 * 0x05 = 0x05
-		 * 0xA5 = 0xA5
+		 * Byte: Tested: 2012-01-20 OK 0x05 = 0x05 0xA5 = 0xA5
```
Comment (Zohhak) [inline src/test/java/com/github/s7connector/test/live/RealPLCTest.java]: revert formatting

File: src/test/java/com/github/s7connector/test/live/RealPLCTest.java [hunk 14]
```diff
@@ -17,145 +17,104 @@
...
 		db.b1 = true;
 		db.b3 = true;
 		db.by1 = 0x5A;
-		
+
 		Date d = new Date();
-		System.out.println(d);
-		System.out.println(d.getTime());
-		
+		LOGGER.trace(d.toString());
```
Comment (Zohhak) [inline src/test/java/com/github/s7connector/test/live/RealPLCTest.java]: toString optional

File: src/test/java/com/github/s7connector/test/live/RealPLCTest.java [hunk 15]
```diff
@@ -17,145 +17,104 @@
...
+		LOGGER.trace(Long.toString(d.getTime()));
```
Comment (Zohhak) [inline src/test/java/com/github/s7connector/test/live/RealPLCTest.java]: toString optional

File: src/test/java/com/github/s7connector/test/live/RealPLCTest.java (hunk 14 above)
Comment (sleim22) [reply src/test/java/com/github/s7connector/test/live/RealPLCTest.java]: LOGGER expects a String and d is a Date

File: src/main/java/com/github/s7connector/impl/nodave/PLCinterface.java [hunk 16]
```diff
@@ -57,7 +61,7 @@ public int read(final byte[] b, int start, int len) {
 					}
 					retry++;
 				} catch (final InterruptedException e) {
-					e.printStackTrace();
+					LOGGER.error("Interrupted ", e);
```
Comment (Zohhak) [inline src/main/java/com/github/s7connector/impl/nodave/PLCinterface.java]: unnecessary whitespace

File: src/main/java/com/github/s7connector/impl/nodave/PLCinterface.java [hunk 17]
```diff
@@ -77,7 +81,7 @@ public void write(final byte[] b, final int start, final int len) {
 		try {
 			this.out.write(b, start, len);
 		} catch (final IOException e) {
-			System.err.println("Interface.write: " + e);
+			LOGGER.error("Interface.write: ", e);
```
Comment (Zohhak) [inline src/main/java/com/github/s7connector/impl/nodave/PLCinterface.java]: unnecessary colon and whitespace

**Review by Zohhak — CHANGES_REQUESTED** [review]

**Review by Zohhak — APPROVED** [review]
